build_calendar.py

```diff
@@ -0,0 +1,118 @@
+    except KeyError:
+        print("Malformed response for {}, {}, {}".format(country, year, calendar_type))
+        return None
+
+def mixin_events(ical, locale):
+    # Global mix ins
+    for event in GlobalHolidays.MIXINS:
+        ical.add_component(event.to_ics())
+
+    # Locale specific mix ins
+    # ...
+
+def build_calendars(locales):
+    try:
+        api_key = os.environ['CALENDARIFIC_API_KEY']
+    except KeyError:
+        sys.exit("No `CALENDARIFIC_API_KEY` defined.")
+
+    try:
+        is_free_tier = os.environ['os.CALENDARIFIC_IS_FREE_TIER'].lower() == 'true'
+    except KeyError:
+        is_free_tier = True
+
+    years_to_generate = 3
+    current_year = datetime.now().year
+    date_span = "{}-{}".format(current_year, current_year + years_to_generate)
+
+    calendar_metadata = []
+
+    print("Querying calendar data from provider")
+
+    for locale, country_name in locales.items():
+        ical = icalendar.Calendar()
+        ical.add('prodid', '-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN')
+        ical.add('version', '2.0')
+
+        mixin_events(ical, locale)
+
+        # Wait 1 second due to free api restrictions
+        if is_free_tier:
+            time.sleep(1)
+
+        for i in range(0, years_to_generate):
+            year = current_year + i
+
+            for calendar_type in [CalendarTypes.NATIONAL, CalendarTypes.OBSERVANCE]:
+                try:
+                    holidays = query_calendarific(api_key, locale, year, calendar_type.value)
+                    formatted_holidays = [Calendarific(holiday, year, calendar_type) for holiday in holidays]
+
+                    for holiday in formatted_holidays:
+                        ical.add_component(holiday.to_ics())
+                except requests.HTTPError as err:
+                    match err.response.status_code:
+                        # API limit reached, only valid for free tier
+                        case requests.status_codes.codes.too_many_requests:
+                            sys.exit("API limit reached")
+                        # Bad API key
+                        case requests.status_codes.codes.unauthorized:
+                            sys.exit("Bad or malformed API key")
+                        case _:
+                            print("Error {}: {}".format(err.response.status_code, err.response.reason))
+                            continue
+                except CountryNotSupportedException as err:
+                    print("Country code {} is not a supported locale.".format(err.country))
+                    continue
+
+        calendar_name = '{}Holidays.ics'.format(country_name)
+
+        calendar_metadata.append({
+            'country': country_name,
+            'filename': "autogen/{}".format(calendar_name),
+            'datespan': date_span,
+            'authors': 'Autogenerated'
+        })
+
+        with open('media/caldata/autogen/{}'.format(calendar_name), 'wb') as fh:
```

Comment:
nit: `open` will error if the directory hasn't been created yet. Since we are going to check calendars into the repository though, the directory should always be there after this patch.